### ISSUE TITLE
Man in middleattack reported 

### ISSUE THREAD

**Issue by harikt** [composer/composer#4806]
I ran into issue when trying to install the dependencies for https://github.com/auraphp/bin . Is there anything I can understand / fix the issue .

Thank you

```
[Composer\Repository\RepositorySecurityException]                            
  The contents of http://packagist.org/p/provider-2013%243f2d811f4f5b35075700  
  c00660c2ed878855cc9532863e8126dd76278ad5da20.json do not match its signatur  
  e. This should indicate a man-in-the-middle attack. Try running composer ag  
  ain and report this if you think it is a mistake. 
```

running on -vvv

```
composer install -vvv
Reading ./composer.json
Loading config file /home/hari/.composer/config.json
Loading config file /home/hari/.composer/auth.json
Loading config file ./composer.json
Executing command (/var/www/github.com/auraphp/bin): git describe --exact-match --tags
Executing command (/var/www/github.com/auraphp/bin): git branch --no-color --no-abbrev -v
Reading /home/hari/.composer/composer.json
Loading config file /home/hari/.composer/config.json
Loading config file /home/hari/.composer/auth.json
Loading config file /home/hari/.composer/composer.json
Loading config file /home/hari/.composer/auth.json
Reading /home/hari/.composer/vendor/composer/installed.json
Loading composer repositories with package information
Downloading https://packagist.org/packages.json
Writing /home/hari/.composer/cache/repo/https---packagist.org/packages.json into cache
Installing dependencies (including require-dev)
Downloading http://packagist.org/p/provider-2013%243f2d811f4f5b35075700c00660c2ed878855cc9532863e8126dd76278ad5da20.json
Downloading http://packagist.org/p/provider-2013%243f2d811f4f5b35075700c00660c2ed878855cc9532863e8126dd76278ad5da20.json
Downloading http://packagist.org/p/provider-2013%243f2d811f4f5b35075700c00660c2ed878855cc9532863e8126dd76278ad5da20.json

                                                                               
  [Composer\Repository\RepositorySecurityException]                            
  The contents of http://packagist.org/p/provider-2013%243f2d811f4f5b35075700  
  c00660c2ed878855cc9532863e8126dd76278ad5da20.json do not match its signatur  
  e. This should indicate a man-in-the-middle attack. Try running composer ag  
  ain and report this if you think it is a mistake.                            
                                                                               

Exception trace:
 () at phar:///var/www/composer.phar/src/Composer/Repository/ComposerRepository.php:623
 Composer\Repository\ComposerRepository->fetchFile() at phar:///var/www/composer.phar/src/Composer/Repository/ComposerRepository.php:516
 Composer\Repository\ComposerRepository->loadProviderListings() at phar:///var/www/composer.phar/src/Composer/Repository/ComposerRepository.php:285
 Composer\Repository\ComposerRepository->whatProvides() at phar:///var/www/composer.phar/src/Composer/DependencyResolver/Pool.php:199
 Composer\DependencyResolver\Pool->computeWhatProvides() at phar:///var/www/composer.phar/src/Composer/DependencyResolver/Pool.php:188
 Composer\DependencyResolver\Pool->whatProvides() at phar:///var/www/composer.phar/src/Composer/DependencyResolver/RuleSetGenerator.php:168
 Composer\DependencyResolver\RuleSetGenerator->whitelistFromPackage() at phar:///var/www/composer.phar/src/Composer/DependencyResolver/RuleSetGenerator.php:322
 Composer\DependencyResolver\RuleSetGenerator->getRulesFor() at phar:///var/www/composer.phar/src/Composer/DependencyResolver/Solver.php:208
 Composer\DependencyResolver\Solver->solve() at phar:///var/www/composer.phar/src/Composer/Installer.php:523
 Composer\Installer->doInstall() at phar:///var/www/composer.phar/src/Composer/Installer.php:232
 Composer\Installer->run() at phar:///var/www/composer.phar/src/Composer/Command/InstallCommand.php:134
 Composer\Command\InstallCommand->execute() at phar:///var/www/composer.phar/vendor/symfony/console/Command/Command.php:259
 Symfony\Component\Console\Command\Command->run() at phar:///var/www/composer.phar/vendor/symfony/console/Application.php:844
 Symfony\Component\Console\Application->doRunCommand() at phar:///var/www/composer.phar/vendor/symfony/console/Application.php:192
 Symfony\Component\Console\Application->doRun() at phar:///var/www/composer.phar/src/Composer/Console/Application.php:167
 Composer\Console\Application->doRun() at phar:///var/www/composer.phar/vendor/symfony/console/Application.php:123
 Symfony\Component\Console\Application->run() at phar:///var/www/composer.phar/src/Composer/Console/Application.php:98
 Composer\Console\Application->run() at phar:///var/www/composer.phar/bin/composer:43
 require() at /var/www/composer.phar:25

install [--prefer-source] [--prefer-dist] [--dry-run] [--dev] [--no-dev] [--no-plugins] [--no-custom-installers] [--no-autoloader] [--no-scripts] [--no-progress] [-v|vv|vvv|--verbose] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--ignore-platform-reqs] [--] [<packages>]...
```

**Comment by harikt** [comment]
Came to know about `composer diagnose` via #2835 . These are the results.

```
Checking composer.json: OK
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com oauth access: OK
Checking disk free space: OK
Checking composer version: OK
```

Cleared all cache . Removed the folder itself `~/.composer` .

Still I cannot install a single package, sadly I am stuck :( .

**Comment by harikt** [comment]
Actually I noticed some of the files where not served from packagist when I tried. Eg : http://packagist.org/p/provider-2013%243f2d811f4f5b35075700c00660c2ed878855cc9532863e8126dd76278ad5da20.json , that returned a 404 . I am not sure why the server did the same, but currently it returns all the json stuff.

I have also seen the same on #4787  like there was no version. Is that anything to do with the server ?
